pkg/apiserver/query/types.go

```diff
@@ -86,34 +68,34 @@ func (p *Pagination) GetPaginationSettings(total int) (startIndex, endIndex int)
 
 func New() *Query {
 	return &Query{
-		Pagination: &Pagination{
-			Limit: -1,
-			Page:  -1,
-		},
-		SortBy:    "",
-		Ascending: false,
-		Filters:   []Filter{},
+		Pagination: NoPagination,
+		SortBy:     "",
+		Ascending:  false,
+		Filters:    []Filter{},
 	}
 }
 
 type Filter struct {
 	Field Field
-	Value Comparable
+	Value Value
```

Comment:
What's the intention of changing Comparable to Value(string)?

Reply:
It is difficult to extend a compareable object, we need to define a lot of uncommon comparable objects under the `pkg/apiserver/query`. Usually we only need to compare based on string type and implement a special compare function for the resource. I implement the general compare function for metav1.ObjectMeta. https://github.com/kubesphere/kubesphere/pull/1989/commits/028d229c083648d24e045730d4e2f2c2061f050b#diff-ba2636a45a25c29c988c339b74061594R72-R120